common/addons/config-noise-function/src/main/java/com/dfsek/terra/addons/noise/samplers/TranslateSampler.java

```diff
@@ -0,0 +1,27 @@
+package com.dfsek.terra.addons.noise.samplers;
+
+import com.dfsek.terra.api.noise.NoiseSampler;
+
+
+public class TranslateSampler implements NoiseSampler {
+    
+    private final NoiseSampler sampler;
+    private final double dx, dy, dz;
+    
+    public TranslateSampler(NoiseSampler sampler, double dx, double dy, double dz) {
+        this.sampler = sampler;
+        this.dx = dx;
+        this.dy = dy;
+        this.dz = dz;
+    }
+    
+    @Override
+    public double noise(long seed, double x, double y) {
+        return sampler.noise(seed, x - dx, y - dz);
+    }
+    
+    @Override
+    public double noise(long seed, double x, double y, double z) {
+        return sampler.noise(seed, x - dx, y - dy, z - dz);
```

Comment:
Subtracting the translation makes sense, e.g. when using DISTANCE sampler, if I want the distance calculation to be translated such that the center (i.e. the smallest distance) is at position (500,0), passing x=500 to the TRANSLATE sampler must mean that x=0 should be passed to the inner DISTANCE sampler. To do so, it would need to subtract the translation, i.e. 500 (world pos) - 500 (dx)